langchain/src/smith/runner_utils.ts

```diff
@@ -33,6 +35,30 @@ export type ChainOrFactory =
   | (() => (obj: unknown) => unknown)
   | (() => (obj: unknown) => Promise<unknown>);
 
+class RunIdResolver {
```

Comment:
If this is a decent abstraction we could factor out and include in the langsmith docs too 🤷

Reply:
I like it! Would just change to `Extractor`